### PR TITLE
add tests for systemd

### PR DESCRIPTION
**Pull request by ramonskie**
this will add tests for systemd.

### CODE REVIEW

**Comment by jpalermo** [comment]
@ramonskie, is the commented code intentional?

File: lib/bat/bosh_helper.rb [hunk 1]
```diff
@@ -102,6 +102,25 @@ def bosh_ssh(job, index, command, options = {})
...
+    end
+
+    def service_command(job, index, options = {})
+      result = bosh_ssh(job, index, 'sudo cat /var/vcap/bosh/agent.json', options)
+      agent_json = result.output.lines.select { |line| line.include?('stdout') }.map { |line| line.split('|').last.strip }.join
+      agent_settings = JSON.load(agent_json)
+      service_manager = agent_settings.dig('Platform', 'Linux', 'ServiceManager')
+
+      if service_manager.nil?
+        command = "sv"
```
Comment (lnguyen) [inline lib/bat/bosh_helper.rb:116]: I would just do return "sv" elseif == systemd return "systemctl" save to simplify code

File: spec/system/service_configuration_spec.rb [hunk 2]
```diff
@@ -278,7 +291,148 @@ def batlight_running_on_instance(ip)
...
+            echo "SUCCESS"
+          EOF
+          output = bosh_ssh(instance_name, instance_id, cmd, deployment: deployment.name).output
+          expect(output).to include("SUCCESS")
+        end
+      end
+    end
+
+    context 'when restarted after agent has been started' do
+      # it 'does not delete /etc/service/monit' do
```
Comment (lnguyen) [inline spec/system/service_configuration_spec.rb:372]: Can we still do this on sv machines? for jammy stemcells

File: spec/system/with_release_stemcell_deployment_spec.rb [hunk 3]
```diff
@@ -64,13 +64,14 @@
...
 
   describe 'logs' do
-    it 'should get agent log' do
-      with_tmpdir do
-        expect(bosh_safe('logs batlight/0 --agent', deployment: deployment.name)).to succeed_with /Downloading resource/
-        files = tar_contents(tarfile)
-        expect(files).to include './current'
-      end
-    end
+    # TODO: not implemented yet as the log location has been changed to /var/log/bosh-agent.log
```
Comment (lnguyen) [inline spec/system/with_release_stemcell_deployment_spec.rb:67]: should we fix this first?

**Review by lnguyen — CHANGES_REQUESTED** [review]

File: spec/system/service_configuration_spec.rb (hunk 2 above)
Comment (ramonskie) [reply spec/system/service_configuration_spec.rb:372]: yes i need to put an systemd condition for that as well.

File: spec/system/with_release_stemcell_deployment_spec.rb (hunk 3 above)
Comment (ramonskie) [reply spec/system/with_release_stemcell_deployment_spec.rb:67]: yes we should fix the agent first before accepting this pr at all

**Review by lnguyen — APPROVED** [review]
LGTM

**Pull request merged**